libfwupd/fwupd-device.c

```diff
@@ -205,14 +226,23 @@ fwupd_device_add_issue(FwupdDevice *self, const gchar *issue)
 	FwupdDevicePrivate *priv = GET_PRIVATE(self);
 	g_return_if_fail(FWUPD_IS_DEVICE(self));
 	g_return_if_fail(issue != NULL);
-	for (guint i = 0; i < priv->issues->len; i++) {
+	for (guint i = 0; priv->issues != NULL && i < priv->issues->len; i++) {
 		const gchar *issue_tmp = g_ptr_array_index(priv->issues, i);
 		if (g_strcmp0(issue_tmp, issue) == 0)
 			return;
 	}
+	fwupd_device_ensure_issues(self);
```

Comment:
how about doing this before the loop?  Then the loop logic can stay as is.

Reply:
Ya, fwupd_device_add_child() is in the same boat. Fixed both.